src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java

```diff
@@ -292,6 +303,46 @@ public void onBlockBreak(BlockBreakEvent e) {
         }
     }
 
+    @EventHandler
+    public void onExperienceReceive(PlayerExpChangeEvent e) {
+        if (e.getAmount() > 0 && Talisman.checkFor(e, SlimefunItems.TALISMAN_WISE)) {
+            e.setAmount(e.getAmount() * 2);
+        }
+    }
+
+    @EventHandler
+    public void onPlayerDeath(PlayerDeathEvent e) {
+        Player player = e.getEntity();
+        DamageCause dmgCause = player.getLastDamageCause().getCause();
+
+        if (dmgCause == DamageCause.VOID && Talisman.checkFor(e, SlimefunItems.TALISMAN_RESURRECTED)) {
+            SlimefunPlugin.runSync(() -> {
+                PaperLib.teleportAsync(player, getSafeRespawnLocation(player));
```

Comment:
Also, is this "safe location" really a safe location? I mean if someone fills that saved location with blocks, then the player will be teleported to that unsafe location anyway, right? If so, we should consider doing some checks before this teleport.

Reply:
Thats a good point, but the player can always choose not to set one, and hell then end up at his bedspawn.
And he can always change the respawn point of his talisman. I dont think we should do extra checks to ensure a safe respawn location.

Reply:
And if you do think it's a necessity, it might be better to make this a public method somewhere so it can be reused for other teleport ocassions

Reply:
> Thats a good point, but the player can always choose not to set one, and hell then end up at his bedspawn.
And he can always change the respawn point of his talisman. I dont think we should do extra checks to ensure a safe respawn location.

Fair enough though. This makes the talisman a little unsafe, but that's the cost I guess :lul: